staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -430,7 +470,7 @@ func LabelsMatch(metric *dto.Metric, labelFilter map[string]string) bool {
 	}
 
 	for labelName, labelValue := range labelFilter {
-		if value, ok := metricLabels[labelName]; !ok || value != labelValue {
+		if value, ok := metricLabels[labelName]; !ok || (value != labelValue && labelValue != "*") {
```

Comment:
The `value` is obtained from `metricLabels[labelName]`, which can never be `*`, right?